YaLinqo/EnumerableGeneration.php

```diff
@@ -67,24 +67,16 @@ public static function from ($source)
             return $source;
         else if (is_array($source))
             $it = new \ArrayIterator($source);
-        elseif ($source instanceof \Iterator)
-            $it = $source;
         elseif ($source instanceof \IteratorAggregate)
             $it = $source->getIterator();
         elseif ($source instanceof \Traversable)
-            $it = self::fromTraversable($source);
```

Comment:
Good change performace-wise, however `Traversable` can be implemented internally, which means that even if it isn't `IteratorAggregate`, it isn't guaranteed to be `Iterator`, so while `Enumerable.getIterator()` method is allowed to return just `Traversable` and no internal code seems to rely on `$iterator` being `Iterator`, this change in general would be breaking, as a user can call `Iterator`'s members at the end of the method chain (`from(...)->getIterator()->next()`).

I don't think anyone actually does this. The code only breaks if LINQ method chain is empty. It can happen, but I'd say the likelyhood of two unlikely events is too unlikely considering YaLinqo's user base.

I think I'll change only minor version.

Reply:
I'm all in for a minor version change. Thanks for the review!